csv/shared/src/test/scala/fs2/data/csv/RowGeneratorTest.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2024 fs2-data Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.data.csv
+
+import cats.data.NonEmptyList
+import weaver.*
+
+object RowGeneratorTest extends SimpleIOSuite {
+
+  pureTest("Emit error on wrong row size (#621)") {
+    val input = List(
+      Row(NonEmptyList.of("a", "b", "c"), Some(1)),
+      Row(NonEmptyList.of("d", "e"), Some(2)),
+      Row(NonEmptyList.of("f", "g", "h", "i"), Some(3)),
+      Row(NonEmptyList.of("j", "k", "l"), Some(4))
+    )
+    val headers = NonEmptyList.of("first", "second", "third")
```

Comment:
```suggestion
    val headers = NonEmptyList.of("first", "second", "third")
    val input = List(
      Row(NonEmptyList.of("a", "b", "c"), Some(1)),
      Row(NonEmptyList.of("d", "e"), Some(2)),
      Row(NonEmptyList.of("f", "g", "h", "i"), Some(3)),
      Row(NonEmptyList.of("j", "k", "l"), Some(4))
    )
```

I find this test case slightly annoying to read, but except for this reordering (so it matches output order) and maybe adding some `unapplySeq` in cats itself, I found no better.

Reply:
One more thought on this snippet (not to be done in this PR though): Maybe a `row"…, …, …"` string interpolator could be nice, especially for our tests?

Reply:
Tried adding `unapplySeq` to cats, unfortunately it doesn't work because `NonEmptyList` is a `case class` and hence has a generated `unapply` that takes precedence. Would need a dedicated extractor object.